docroot/sites/international.uiowa.edu/modules/international_migrate/src/Plugin/migrate/source/Articles.php

```diff
@@ -99,14 +186,651 @@ public function prepareRow(Row $row) {
+      ]);
+    }
+    else {
+      $this->logger->notice('Unable to update node links: @nids', [
+        '@nids' => implode(', ', $to_update),
+      ]);
+    }
+    // Report possible broken links after our known high water mark
+    // of articles in which we fixed links.
+    $this->reportPossibleLinkBreaks(['node__body' => ['body_value']], 11901);
+  }
+
+  /**
+   * Replace links for several specified nodes.
+   */
+  private function replaceSpecifiedLinks($nids) {
+    $db = \Drupal::database();
+    if (!$db->schema()->tableExists('migrate_map_international_articles')) {
+      return FALSE;
+    }
+    $map = $db->select('migrate_map_international_articles', 'm')
+      ->fields('m', ['sourceid1', 'destid1'])
+      ->execute()
+      ->fetchAllKeyed();
+
+    $entity_manager = \Drupal::service('entity_type.manager')
+      ->getStorage('node');
+    $block_manager = \Drupal::service('entity_type.manager')
+      ->getStorage('block_content');
+    $nodes = $entity_manager->loadMultiple($nids);
+    foreach ($nodes as $node) {
+      // Grab our section from the node's layout. Here we know
+      // the structure of our pages, so we can grab it directly.
+      $layout = $node->get('layout_builder__layout');
+      $section = $layout->get(2)->getValue()['section'];
+      $section_array = $section->toRenderArray();
+      $uuid = array_keys($section_array['content'])[0];
+      $component = $section_array['content'][$uuid];
+      $revision_id = $component['#configuration']['block_revision_id'];
+      $block = $block_manager->loadRevision($revision_id);
+      $block_text = $block->field_uiowa_text_area->value;
+
+      $doc = Html::load($block_text);
+      // Parse links.
+      $links = $doc->getElementsByTagName('a');
+      $i = $links->length - 1;
+      while ($i >= 0) {
+        $link = $links->item($i);
+        $href = $link->getAttribute('href');
+        if (strpos($href, '/node/') === 0 || stristr($href, 'international.uiowa.edu/node/')) {
+          $nid = explode('node/', $href)[1];
+          if ($lookup = $map[$nid]) {
+            $link->setAttribute('href', '/node/' . $lookup);
+            $link->parentNode->replaceChild($link, $link);
+          }
+          else {
+            $text = $doc->createTextNode($link->nodeValue);
+            $link->parentNode->replaceChild($text, $link);
+          }
+        }
+        $i--;
+      }
+      // Re-serialize the DOM and set into the body text.
+      $block_text = Html::serialize($doc);
+      $block->field_uiowa_text_area->value = $block_text;
+      $block->save();
+      // Set the new revision in the component
+      // and place it back into the section array.
+      $component['#configuration']['block_revision_id'] = $block->getRevisionId();
+      $section_array['content'][$uuid] = $component;
+      // Remove the old section and append our new one.
+      $layout->removeSection(2);
+      $layout->appendSection(Section::fromArray($section_array));
```

Comment:
This fromArray method isn't setting the layoutId, though there is one in the array...need to work out how to make sure this gets set, otherwise it's leading the the `the "" plugin does not exist` error when attempting to save the node.